src/Temperature-Accretech_UFseries/main.py

```diff
@@ -0,0 +1,160 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# SweepMe! device class
+# Type: Temperature
+# Device: Accretech UF series
+from __future__ import annotations
+
+from pysweepme.EmptyDeviceClass import EmptyDevice
+from pysweepme.FolderManager import addFolderToPATH
+
+addFolderToPATH()
+
+# standard import
+# import accretech_uf
+# importlib.reload(accretech_uf)
+
+# direct import by path
+import imp
+import os
+
+accretech_uf = imp.load_source(
+    "accretech_uf",
+    os.path.dirname(os.path.abspath(__file__)) + os.sep + r"libs\accretech_uf.py",
+)
+
+# this is needed as a fallback solutions as pysweepme.UserInterface is not available for all 1.5.5 update versions
+try:
+    from pysweepme.UserInterface import message_box
+except ModuleNotFoundError:
+    message_box = print
+    print(
+        "Accretech driver: Please use the latest version of SweepMe! to use this "
+        "driver to display a message box. "
+        "Fallback to 'print' method.",
+    )
+
+
+class Device(EmptyDevice):
+    def __init__(self):
+        EmptyDevice.__init__(self)
+
+        self.instance_key = None
+        self.port_str = None
+        self.prober: accretech_uf.AccretechProber = None
+        self.temperature: float = 0.0
+
+        self.shortname = "Accretech"
+        self.variables = ["Temperature"]
+        self.units = [""]  # unit of X and Y is either µm or inch, depending on system settings
+        self.plottype = [True]
+        self.savetype = [True]
+
+        self.port_manager = True
+        self.port_types = ["GPIB"]
+        self.port_properties = {
+            "timeout": 5,  # only for write&read messages, SRQ&STB commands have own timeout
+            "GPIB_EOLwrite": "\r\n",
+            "GPIB_EOLread": "\r\n",
+        }
+
+        self.verbosemode = True
+
+    def set_GUIparameter(self) -> dict:  # noqa: N802
+        """Set initial GUI parameters."""
+        return {
+            "SweepMode": ["None", "Temperature"],
+            "Temperature unit": ["°C"],
+        }
+
+    def get_GUIparameter(self, parameter: dict) -> None:  # noqa: N802
+        """Handle SweepMe GUI parameters."""
+        self.units[0] = parameter["Temperature unit"]
+
+        self.port_str = parameter["Port"]
+
+    def connect(self) -> None:
+        """Connect to the device and handle multiple instances."""
+        # creating an AccretechProber instance that handles all communication
+        # TODO: Handle multiple instances
+        self.instance_key = f"Accretech_UF_{self.port_str}"
+
+        if self.instance_key in self.device_communication:
+            self.port = self.device_communication[self.instance_key]
+        else:
+            self.device_communication[self.instance_key] = self.port
```

Comment:
This section could be commented out for now. The GPIB port itself is handled by SweepMe's port manager, so that each driver instance automatically gets the same GPIB port. 
However, it might be that we need the self.device_communication to mark whether the SRQ event is already registered or not.
If this can be done only once, we probably would need to also revise the accretech_uf library because it would need a switch to decide whether it should register for the SRQ event during connect. But this is something that we can check once the driver is tested.